server/covmanager/templates/collections/index.html

```diff
@@ -229,11 +231,67 @@
         this.suggestions[key].value = json["results"]
       })
     }, 500),
-		take_suggestion: function(key, val) {
-			this.suggestions[key].enabled = false
-			this.suggestions[key].value = []
-			this.search[key].value = val
-		}
+    take_suggestion: function(key, val) {
+      this.suggestions[key].enabled = false
+      this.suggestions[key].value = []
+      this.search[key].value = val
+    },
+    aggregate: function() {
+      let ids = []
+      for (let i = 0; i < this.ordered_collections.length; ++i) {
+        ids.push(this.ordered_collections[i].id)
+      }
+
+      if (ids) {
+        sweetAlert({
+          title: "Create Collection Aggregation",
+          text: 'Enter optional new description for result collection',
+          type: 'input',
+          showCancelButton: true,
+          animation: "slide-from-top"
+        }, function(description){
+          if (description === false) {
+            /* User pressed cancel button */
+            return;
+          }
+
+          // Call API, get new collection id back, then navigate to it
+          this.loading = true
+
+          let data = {
+            'ids': ids.join(",")
+          }
+          if (description) {
+            data['description'] = description
+          }
+
+          fetch(URLS.aggregate, {
+            method: 'POST',
+            credentials: 'same-origin',
+            headers: {
+              "Content-Type": "application/json; charset=utf-8",
+              "X-Requested-With": "XMLHttpRequest"
+            },
+            body: JSON.stringify(data),
+          })
+          .then(response => {
+            this.loading = false
+            if (response.ok) {
+              return response.json()
+            }
+            sweetAlert('Oops', E_SERVER_ERROR, 'error')
+          })
+          .then(json => {
+            let newid = json["newid"]
+            if (newid) {
+              window.open(json["newid"] + "/browse", '_self');
+            } else {
+              sweetAlert('Oops', E_SERVER_ERROR, 'error')
+            }
+          })
```

Comment:
Is `.catch()` needed here in case we get an http error?

Reply:
Afaik, it should not be because fetch() does not raise on an HTTP error but instead returns the response but response.ok won't evaluate to true.